src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -439,6 +434,11 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
             isExternal: isExternal,
             isProxied: isProxied);
 
+        if (builder.Resource.Annotations.OfType<EndpointAnnotation>().Any(sb => string.Equals(sb.Name, annotation.Name, StringComparisons.EndpointAnnotationName)))
+        {
+            throw new DistributedApplicationException($"Endpoint with name '{annotation.Name}' already exists");
```

Comment:
I think we should improve the error message. We default the scheme to the name so they may not even know the endpoint has a name. We should try out best to tell them that it does and that the if no name is specified, the scheme will be used as the name. Then we should recommend that they specify a unique name per endpoint.